src/axom/quest/tests/quest_intersection_shaper.cpp

```diff
@@ -0,0 +1,553 @@
+//#define GENERATE_BASELINES
+
+// Uncomment this macro to save MFEM datasets for use in VisIt.
+//#define VISUALIZE_DATASETS
+
+std::vector<std::string> case1 {"shaping/case1/case1_012.yaml",
+                                "shaping/case1/case1_021.yaml",
+                                "shaping/case1/case1_102.yaml",
+                                "shaping/case1/case1_120.yaml",
+                                "shaping/case1/case1_201.yaml",
+                                "shaping/case1/case1_210.yaml"};
+
+std::vector<std::string> case2 {"shaping/case2/case2_012.yaml",
+                                "shaping/case2/case2_021.yaml",
+                                "shaping/case2/case2_102.yaml",
+                                "shaping/case2/case2_120.yaml",
+                                "shaping/case2/case2_201.yaml",
+                                "shaping/case2/case2_210.yaml"};
+
+std::vector<std::string> case3 {"shaping/case3/case3_012.yaml",
+                                "shaping/case3/case3_021.yaml",
+                                "shaping/case3/case3_102.yaml",
+                                "shaping/case3/case3_120.yaml",
+                                "shaping/case3/case3_201.yaml",
+                                "shaping/case3/case3_210.yaml"};
+
+std::vector<std::string> case4 {"shaping/case4/case4.yaml",
+                                "shaping/case4/case4_overwrite.yaml"};
+
+using std::cout;
+using std::endl;
+
+namespace quest = axom::quest;
+namespace slic = axom::slic;
+namespace sidre = axom::sidre;
+namespace klee = axom::klee;
+
+std::string pjoin(const std::string &path, const std::string &filename)
+{
+  return axom::utilities::filesystem::joinPath(path, filename);
+}
+
+void psplit(const std::string &filepath, std::string &path, std::string &filename)
+{
+  axom::Path p(filepath);
+  path = p.dirName();
+  filename = p.baseName();
+}
+
+std::string dataDirectory() { return AXOM_DATA_DIR; }
+
+std::string testData(const std::string &filename)
+{
+  return pjoin(dataDirectory(), filename);
+}
+
+std::string baselineDirectory()
+{
+  return pjoin(pjoin(pjoin(dataDirectory(), "quest"), "regression"),
+               "quest_intersection_shaper");
+}
+
+std::string yamlRoot(const std::string &filepath)
+{
+  std::string retval, path, filename;
+  psplit(filepath, path, filename);
+  auto idx = filename.rfind(".");
+  if(idx != std::string::npos)
+    retval = filename.substr(0, idx);
+  else
+    retval = filename;
+  return retval;
+}
+
+mfem::GridFunction *newGridFunction(mfem::Mesh *mesh)
+{
+  const int vfOrder = 0;
+  const int dim = mesh->Dimension();
+  mfem::L2_FECollection *coll =
+    new mfem::L2_FECollection(vfOrder, dim, mfem::BasisType::Positive);
+  mfem::FiniteElementSpace *fes = new mfem::FiniteElementSpace(mesh, coll);
+  mfem::GridFunction *gf = new mfem::GridFunction(fes);
+  gf->MakeOwner(coll);
+  memset(gf->GetData(), 0, gf->Size() * sizeof(double));
```

Comment:
I changed the code to the MFEM way - and added a comment (because I am uninitiated and it reminds me of gf = 0 which would be setting the *pointer* to NULL).